flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/operations/ddl/AlterOperation.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.operations.ddl;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.table.operations.Operation;
+
+/**
+ * A {@link Operation} that describes the DDL statements, e.g. ALTER TABLE or ALTER DATABASE.
+ *
+ * <p>Different sub operations can have their special instances. For example, a
+ * alter table operation will have a {@link org.apache.flink.table.catalog.CatalogTable} instance,
+ * while a alter database operation will have a
+ * {@link org.apache.flink.table.catalog.CatalogDatabase} instance.
```

Comment:
It seems to me this is just a marker interface, whether/what instance is used should be up to implementing classes.